src/sentry/static/sentry/app/components/substances/assignToWorkflow.jsx

```diff
@@ -139,23 +152,31 @@ const AssignToWorkflowButton = createReactClass({
   },
 
   render() {
-    if (this.props.processDefinition.loading) {
+    if (this.props.process.processDefinitionLoading) {
       return <LoadingIndicator />;
     }
 
-    const presets = Object.entries(this.props.processDefinition.presetsById).map(
+    const presets = Object.entries(this.props.presetsById).map((entry) => {
+      return {value: entry[1].id, label: entry[1].name};
+    });
+    const presetsSorted = sortBy(presets, 'label');
+
+    // TODO: Here we should handle the case where two workflows have the same name (but different
+    // namespaces), by showing the full name in that case
+    const processDefinitions = Object.entries(this.props.processDefinition.byIds).map(
       (entry) => {
-        return {value: entry[0], label: entry[1].name};
+        const processId = entry[1].id;
+        const elements = processId.split('.');
```

Comment:
I just want to check, should the process name be parsed out from process-id, and not process full name?

Reply:
Ahh, right. Good catch. It's correct in this case as the ID here contains the name if I'm not mistaken. But it looks funky and just getting it from the name would be better :) A thing for the refactoring process!